internal/executors/action/action.go

```diff
@@ -41,16 +43,17 @@ type IExecuter interface {
 type Executor struct {
 	capabilities map[string]capability.ICapability
 	reporter     reporter.IStepReporter
+	time         timeUtil.ITime
 }
 
 func (executor *Executor) Execute(meta execution.Metadata, metadata PlaybookStepMetadata) (cacao.Variables, error) {
 
-	executor.reporter.ReportStepStart(meta.ExecutionId, metadata.Step, metadata.Variables)
+	executor.reporter.ReportStepStart(meta.ExecutionId, metadata.Step, metadata.Variables, executor.time.Now())
 
 	if metadata.Step.Type != cacao.StepTypeAction {
 		err := errors.New("the provided step type is not compatible with this executor")
 		log.Error(err)
-		executor.reporter.ReportStepEnd(meta.ExecutionId, metadata.Step, cacao.NewVariables(), err)
+		executor.reporter.ReportStepEnd(meta.ExecutionId, metadata.Step, cacao.NewVariables(), err, executor.time.Now())
```

Comment:
Try to use `defer` it might be that you need to move up the creation of the `returnVariables`